js/tabs.js

```diff
@@ -465,26 +469,13 @@ Tabs.prototype.saveEntry_ = function(tab, entry, opt_callback) {
   this.save(tab, opt_callback);
 };
 
-Tabs.prototype.onDocChanged_ = function(e, session) {
+Tabs.prototype.onDocChanged_ = function(_, { type }) {
+  // Assume that updates are coming from the current tab, if they arn't
+  // we have no real way to recover so failing here is acceptable.
   var tab = this.currentTab_;
-  if (this.currentTab_.getSession() !== session) {
-    console.warn('Something wrong. Current session should be',
-                 this.currentTab_.getSession(),
-                 ', but this session was changed:', session);
-    for (var i = 0; i < this.tabs_; i++) {
```

Comment:
why is the previous error handling no longer useful?

Reply:
i originally felt that it was doing nothing helpful since if the tabs state is corrupted switching the current tab to whatever is generating events sounds like it wouldn't really help address the root problem. However in writing up a justification and reading over it i think you are right in that there is no ham to keeping it there and trying to reset the tabs state on a weird event might make more sense then not. Will put it back in.

Reply:
This actually doesn't do any error handling it just logs the fact that something went wrong, will put it back in since in reading over it it's doing more good then bad

Reply:
Ack.